plugin.audio.tilos/addon.py

```diff
@@ -0,0 +1,507 @@
+from shutil import rmtree, copy
+import traceback
+from pprint import pprint
+import json
+import time
+import calendar
+import datetime
+import re
+
+
+############################################
+# Define required statics
+############################################
+
+__plugin__ = "Tilos"
+__version__ = '0.0.5'
+__author__ = 'Gabor Boros'
+__date__ = '2014-07-28'
+__addon__ = xbmcaddon.Addon()
+__addonname__ = __addon__.getAddonInfo('name')
+
+BASE_URL = 'https://tilos.hu'
+BASE_URL_PAGE = 'https://tilos.hu/page'
+BASE_URL_SHOWS = BASE_URL + '/api/v1/show'
+BASE_URL_EPISODES = BASE_URL_SHOWS + '/%s/episodes?start=%d&end=%d'
+BASE_URL_EPISODES_BY_DATE = BASE_URL + '/api/v1/episode?start=%d&end=%d'
+BASE_URL_SOUNDSTORE = BASE_URL + '/api/v1/mix'
+HEADERS = {'User-Agent' : 'XBMC Plugin v0.0.4'}
+LIVE_URL_256 = 'http://stream.tilos.hu/tilos'
+LIVE_URL_128 = 'http://stream.tilos.hu/tilos_128.mp3'
+
+dialogProgress = xbmcgui.DialogProgress()
+dialog = xbmcgui.Dialog()
+
+base_url = sys.argv[0]
+addon_handle = int(sys.argv[1])
+args = urlparse.parse_qs(sys.argv[2][1:])
+
+xbmcplugin.setContent(addon_handle, 'songs')
+Addon = xbmcaddon.Addon("plugin.audio.tilos")
+
+mode = args.get('mode', None)
+
+debug = False
+
+############################################
+# Functions
+############################################
+
+
+def log(msg):
+
+    if (debug):
+        if type(msg) not in (str, unicode):
+            xbmc.log("[%s]: %s" % (__plugin__, type(msg)))
+            pprint(msg)
+        else:
+            if type(msg) in (unicode,):
+                msg = msg.encode('utf-8')
+            xbmc.log("[%s]: %s" % (__plugin__, msg))
+
+
+def buildURL(query):
+    return base_url + '?' + urllib.urlencode('' if query is None else query)
+
+
+def getURL(url):
+    log(' > getURL(%s)' % (url))
+    req = Request(url, None, HEADERS)
+     
+    try:
+        response = urlopen(req)
+     
+    except URLError, e:
+     
+        if hasattr(e, 'reason'):
+            log(getString(30003))
+            log(getString(30004) + str(e.reason))
+            xbmcgui.Dialog().ok(__addonname__, getString(30003), '', getString(30004) + str(e.reason))
+     
+        elif hasattr(e, 'code'):
+            log(getString(30005))
+            log(getString(30006) + str(e.code))
+            xbmcgui.Dialog().ok(__addonname__, getString(30005), '', getString(30006) + str(e.code))
+    
+    except Exception, e:
+        
+        if hasattr(e, 'reason'):
+            log(e.reason)
+        
+    else:
+        return response.read()
+        
+    
+def getString(stringID):
+    return getUString(__addon__.getLocalizedString(stringID))
+    
+
+def getUString(string):
+    if string is None:
+        return ''
+    else:
+        return string.encode('utf8')
+
+
+def listRootMenu():
+    log(' > listRootMenu()')
+
+    name = getCurrentShowName()
+    li = xbmcgui.ListItem('%s [I]%s[/I]' % (getString(30007), getUString(name)), thumbnailImage=name)
```

Comment:
In your next update please remove the inits for iconImage and thumbnailImage from ListItem, they are now deprecated.  Please use ListItem.setArt() instead.